server.go

```diff
@@ -442,13 +446,14 @@ func (p *Proxy) Handle(conn *net.TCPConn, connectionID string) {
 		glog.V(7).Infof("[%s] Error sending data to backend: %v", connectionID, err)
 		return
 	}
-	Copycat(clientConn, conn, connectionID)
+	metrics.AddBackendBytesSent(proxyIngressBackend, int64(n))
+	Copycat(clientConn, conn, connectionID, proxyIngressBackend)
 }
 
-func Copycat(client *net.TCPConn, server *net.TCPConn, connectionID string) {
+func Copycat(client *net.TCPConn, server *net.TCPConn, connectionID string, backend string) {
 	glog.V(6).Infof("[%s] Initiating copy between %s and %s", connectionID, client.RemoteAddr().String(), server.RemoteAddr().String())
 
-	doCopy := func(s, c *net.TCPConn, cancel chan<- string) {
+	doCopy := func(s, c *net.TCPConn, cancel chan<- string, receiving bool) {
```

Comment:
Instead of passing a bool, would it be possible to pass a function pointer with the signature (backend string, numWritten int64)? This way doCopy() doesn't need an if-then-else statement. The two doCopy() invocations could then pass metrics.AddBackendBytesRcvd and metrics.AddBackendBytesSent as pointers. If this works, it'll simplify the code.